### PR TITLE
fix(operator): Fix operator release pipeline warnings

### PR DESCRIPTION
**Pull request by periklis**
**What this PR does / why we need it**:
Fixes warnings as mentioned in the latest release pipeline reports:
- k8s-operator-hub/communit-operators: [0.7.0](https://github.com/k8s-operatorhub/community-operators/pull/5249#issuecomment-2461701309)
- redhat-openshift-ecosystem/community-operators-prod: [0.7.0](https://github.com/redhat-openshift-ecosystem/community-operators-prod/pull/5460#issuecomment-2462091908)

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:
- Leaving the migration to File-Based-Catalog for a separate PR as per instructions need careful considerations:  https://redhat-openshift-ecosystem.github.io/operator-pipelines/users/fbc_onboarding/
- To run the redhat-openshift-ecosystem static tests please consider following the [official instructions](https://redhat-openshift-ecosystem.github.io/operator-pipelines/users/static_checks/#check_operator_name-warning) on a local fork of the repository by copying the `community-openshift` bundle contents in a new fake version locally.

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [x] Title matches the required conventional commits format, see [here](https://www.conventionalcommits.org/en/v1.0.0/)
  - **Note** that Promtail is considered to be feature complete, and future development for logs collection will be in [Grafana Alloy](https://github.com/grafana/alloy). As such, `feat` PRs are unlikely to be accepted unless a case can be made for the feature actually being a bug fix to existing behavior.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

File: operator/bundle/community-openshift/manifests/loki-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -1771,7 +1771,7 @@ spec:
           - update
           - watch
         - apiGroups:
-          - policy
+          - policy/v1
```
Comment (JoaoBraveCoding) [inline operator/bundle/community-openshift/manifests/loki-operator.clusterserviceversion.yaml:1774]: I believe policy only became stable with `1.21` do we should bump our compatibility to `1.21` from `1.16`

**Review by JoaoBraveCoding — COMMENTED** [review]
Overall lgtm just one comment/question

File: operator/bundle/community-openshift/manifests/loki-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (periklis) [reply operator/bundle/community-openshift/manifests/loki-operator.clusterserviceversion.yaml:1774]: You mean `minKubeVersion`?

Comment (JoaoBraveCoding) [reply operator/bundle/community-openshift/manifests/loki-operator.clusterserviceversion.yaml:1774]: I didn't realise `minKubeVersion` I actually meant `docs/operator/compatibility.md` that still references 1.16

**Review by JoaoBraveCoding — APPROVED** [review]
Ran static tests and deployed community version ✅ 
lgtm

**Pull request merged**